iree/compiler/Conversion/LinalgToLLVM/PadLinalgWorkgroupTiles.cpp

```diff
@@ -106,20 +95,19 @@ class MatmulWorkgroupTilesPadding : public OpRewritePattern<linalg::MatmulOp> {
         rhs.getDefiningOp<linalg::PadTensorOp>())
       return failure();
 
-    auto lhsFullSize = getFullSize(lhs);
-    auto rhsFullSize = getFullSize(rhs);
+    auto workgroupTileSizes = getTileSizes(
+        matmulOp, static_cast<unsigned>(TilingLevel::WorkGroupTiles));
+    auto vectorTileSizes =
+        getTileSizes(matmulOp, static_cast<unsigned>(TilingLevel::Level2Tiles));
+    if (workgroupTileSizes.empty() || vectorTileSizes.empty()) return failure();
 
-    if (lhsFullSize.empty() || rhsFullSize.empty()) return failure();
+    auto lhsFullSize = getUntiledShape(lhs);
+    auto rhsFullSize = getUntiledShape(rhs);
```

Comment:
Fail if either size < 2

Reply:
I dont follow this either. you mean `lhsFullSize.size() < 2` or `rhsFullSize.size() < 2` . This is a matmul. This is size 2 by construction.

Reply:
looking at getUntiledShape will return `vector<2>` or empty, lets keep this ?
```c++
if (lhsFullSize.empty() || rhsFullSize.empty()) return failure();
```